litgpt/lora.py

```diff
@@ -338,8 +340,17 @@ def zero_pad(self, x: torch.Tensor) -> torch.Tensor:
         # Then x has embeddings_size of 256 (2 * 128 as enable_lora only for query and value, not keys) and expected
         # embeddings_size is 384 (self.linear.out_features), so that means that we need to pad from 256 to 384 with zeros, but
         # only for key updates (this is where self.lora_ind comes in handy)
-        result = x.new_zeros(*x.shape[:-1], self.linear.out_features)  # (64, 64, 384)
-        return result.index_copy_(dim=-1, index=self.lora_ind, source=x)  # (64, 64, 384)
+
+        if not self.mps_compatibility_mode:
+            result = x.new_zeros(*x.shape[:-1], self.linear.out_features)  # (64, 64, 384)
+            return result.index_copy_(dim=-1, index=self.lora_ind, source=x)  # (64, 64, 384)
+        else:
+            if all(self.enable_lora):
```

Comment:
That shouldn't be in the compatibility mode.
In fact, only index copy should be altered with slicing in the compatibility mode.